nltk/translate/rouge_score.py

```diff
@@ -0,0 +1,207 @@
+'''ROUGE score implementation
+Link to paper:
+http://www.aclweb.org/anthology/W04-1013
+'''
+
+from nltk.util import ngrams
+from nltk.tokenize import TreebankWordTokenizer, PunktSentenceTokenizer
+from nltk import skipgrams
+from nltk.translate.util import jacknifing, lcs
+import numpy as np
+
+
+tokenizer = TreebankWordTokenizer()
+sentence_tokenizer = PunktSentenceTokenizer()
+
+
+def get_score(r_lcs, p_lcs, beta):
+    '''This is a scoring function implementing the
+    F-measure type formula present in the paper.
+    Here:
+    beta = parameter
+    r_lcs = recall factor
+    p_lcs = precision factor
+    '''
+    try:
+        score = ((1+beta**2)*r_lcs*p_lcs)/(r_lcs+(beta**2)*p_lcs)
+    except ZeroDivisionError as e:
+        score = 0
+    return score
+
+
+def rouge_n(references, candidate, n, averaging=True):
+    ''' It is a n-gram recall between a candidate summary
+    and a set of reference summaries.
+    references = list of references
+    candidate =  the candidate string
+    n = the n parameter in  ngram
+    ng_cand = list of ngrams in candidate
+    ng_ref = list of ngrams in reference
+    r_lcs = recall factor
+    p_lcs = precision factor
+    rouge_recall = list containing all the rouge-n scores for
+                   every reference against the candidate
+    '''
+    ng_cand = list(ngrams(tokenizer.tokenize(candidate), n))
+    rouge_recall = []
+    for ref in references:
+        count = 0
+        ng_ref = list(ngrams(tokenizer.tokenize(ref), n))
+        for ngr in ng_cand:
+            if ngr in ng_ref:
+                count += 1
```

Comment:
This inner loop can be optimized by `collections.Counter` intersection functions like how it's done in `bleu_score.py`